### ISSUE TITLE
please create new module service_stat

### ISSUE THREAD

**Issue by ansibot** [ansible/ansible#29690]
From @sharuzzaman on 2016-08-26T10:43:57Z

##### ISSUE TYPE

- Feature Idea
##### COMPONENT NAME

service_stat
##### ANSIBLE VERSION

```
ansible 2.1.0.0
  config file = /home/ansible/.ansible.cfg
  configured module search path = Default w/o overrides
```
##### CONFIGURATION

no
##### OS / ENVIRONMENT

ansible server: CentOS 7
managing:
CentOS 5,6,7
RHEL5
Ubuntu
Debian 6,7,8
##### SUMMARY

Please create service_stat that will return states of the services, such as installed, exist, running, uid of service, pid of service, etc
Right now, there is no cross-platform method to determine of service is running, and the obvious way to cross platform is by using a command module and getting the return value
##### STEPS TO REPRODUCE

```
- name: check service exist or running
  service_stat: name=httpd 
  register: _srv_httpd

- name: restart if httpd service exist or running
  service: name=httpd state=restarted
  when: _srv_httpd.stat.exists == True or _srv_httpd.stat.running == True

```

##### EXPECTED RESULTS

the service status will be checked and returned into a register.
other task can be executed or not based on the status of the register
##### ACTUAL RESULTS

Right now, there is no cross-platform method to determine of service is running or exist or installed, and the obvious way to cross platform is by using a command module and getting the return value

See the example of pain of user having to determine if service exist here: http://stackoverflow.com/questions/30328506/check-if-service-exists-with-ansible

```

```

Copied from original issue: ansible/ansible-modules-core#4555

**Comment by ansibot** [comment]
From @ansibot on 2016-08-26T10:43:57Z



@grossws ping, this issue is waiting for your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
From @dagwieers on 2016-08-26T10:43:57Z

The whole point of [idempotence](https://en.wikipedia.org/wiki/Idempotence#Computer_science_meaning) is that you shouldn't care whether the service is running or not (or what the PID is), but you only declare what you want, e.g. **service: name=httpd state=started**. If it is not running, it will ensure it is running (**[changed]**), if it was already running it doesn't need to do anything (**[ok]**).

So what is missing from this is to know in advance one what systems httpd should be running/restarted, and that's what you do through the use of inventory/groups/roles.

``` yaml
- hosts: webservers
  tasks:
  - name: Ensure httpd is restarted
    service: name=httpd state=restarted
```

To summarize, this feature is not needed if you use Ansible the way it was designed. However there is nothing preventing you from creating a module/plugin that does exactly what you need for your specific use case. wontfix

**Comment by ansibot** [comment]
From @ansibot on 2016-08-26T10:43:57Z

@ansible, ping. This issue is waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)
